programs/clearing_house/src/state/market.rs

```diff
@@ -85,61 +87,67 @@ impl BankBalance for PNLPool {
 #[derive(Default)]
 #[repr(packed)]
 pub struct AMM {
+    // oracle
     pub oracle: Pubkey,
     pub oracle_source: OracleSource,
+    pub last_oracle_price: i128,
+    pub last_oracle_conf_pct: u64,
+    pub last_oracle_delay: i64,
+    pub last_oracle_normalised_price: i128,
+    pub last_oracle_price_twap: i128,
+    pub last_oracle_price_twap_ts: i64,
+    pub last_oracle_mark_spread_pct: i128,
+
     pub base_asset_reserve: u128,
     pub quote_asset_reserve: u128,
+    pub sqrt_k: u128,
+    pub peg_multiplier: u128,
+
     pub terminal_quote_asset_reserve: u128,
-    pub cumulative_repeg_rebate_long: u128,
-    pub cumulative_repeg_rebate_short: u128,
-    pub cumulative_funding_rate_long: i128,
-    pub cumulative_funding_rate_short: i128,
-    pub cumulative_funding_rate_lp: i128,
+    pub net_base_asset_amount: i128,
+    pub quote_asset_amount_long: u128,
+    pub quote_asset_amount_short: u128,
+
+    // funding
     pub last_funding_rate: i128,
     pub last_funding_rate_ts: i64,
     pub funding_period: i64,
-    pub last_oracle_price_twap: i128,
+    pub cumulative_funding_rate_long: i128,
+    pub cumulative_funding_rate_short: i128,
+    pub cumulative_funding_rate_lp: i128,
+    pub cumulative_repeg_rebate_long: u128,
+    pub cumulative_repeg_rebate_short: u128,
+
+    pub mark_std: u64,
     pub last_mark_price_twap: u128,
     pub last_mark_price_twap_ts: i64,
-    pub sqrt_k: u128,
-    pub peg_multiplier: u128,
-    pub total_fee: u128,
-    pub total_fee_minus_distributions: u128,
-    pub total_mm_fee: u128,
-    pub total_exchange_fee: u128,
-    pub net_revenue_since_last_funding: i64,
-    pub total_fee_withdrawn: u128,
+
+    // trade constraints
     pub minimum_quote_asset_trade_size: u128,
-    pub last_oracle_price_twap_ts: i64,
-    pub last_oracle_normalised_price: i128,
-    pub last_oracle_price: i128,
-    pub last_oracle_conf: u64,
-    pub last_oracle_delay: i64,
-    pub last_oracle_mark_spread_pct: i128,
     pub minimum_base_asset_trade_size: u128,
-    pub base_spread: u16,
 
+    // market making
+    pub base_spread: u16,
     pub long_spread: u128,
     pub short_spread: u128,
-
     pub last_bid_price_twap: u128,
     pub last_ask_price_twap: u128,
-    pub net_base_asset_amount: i128,
-    pub quote_asset_amount_long: u128,
-    pub quote_asset_amount_short: u128,
-    pub mark_std: u64,
 
-    pub long_intensity_time: i64,
     pub long_intensity_count: u16,
     pub long_intensity_volume: u64,
-
-    pub short_intensity_time: i64,
     pub short_intensity_count: u16,
     pub short_intensity_volume: u64,
-
-    // upgrade-ability
     pub curve_update_intensity: u8,
 
+    // fee tracking
+    pub total_fee: u128,
+    pub total_mm_fee: u128,
+    pub total_exchange_fee: u128,
+    pub total_fee_minus_distributions: u128,
+    pub total_fee_withdrawn: u128,
+    pub net_revenue_since_last_funding: i64,
+    pub pnl_pool: PNLPool,
```

Comment:
i think the amm and market both having a pnl_pool property is confusing, should have two different ones. We could just have a `deposit_balance` property on market that keeps track of how much of the bank balance it has? the pnl pool might also make more sense being on the market struct instead of the amm struct

Reply:
so have it be a bank_balance type still? that will make the functions calls easy and just rename it?